.github/workflows/rhub.yaml

```diff
@@ -0,0 +1,119 @@
+# This workflow is based on R-hub's generic GitHub Actions workflow file.
+# Its canonical location is at
+# https://github.com/r-hub/actions/blob/main/workflows/rhub.yaml
+
+name: R-hub 🌐
+
+on:
+  workflow_call:
+    inputs:
+      config:
+        description: 'A comma separated list of R-hub platforms to use.'
+        type: string
+        # These platforms correspond roughly to CRAN package checks.
+        # More information: https://github.com/insightsengineering/idr-tasks/issues/781
+        # All supported R-hub platforms can be viewed by running: rhub::rhub_platforms()
+        # Container-based platforms (as opposed to VM-based Windows and macOS platforms)
+        # can be also viewed here: https://r-hub.github.io/containers/containers.html
+        default: 'ubuntu-clang,ubuntu-gcc12,gcc13,ubuntu-next,ubuntu-release,macos-arm64,macos,windows,gcc14,atlas,valgrind,intel,nosuggests,mkl'
```

Comment:
Opening a separate thread to keep the discussion well organised.
This is the results for `teal`:
![image](https://github.com/insightsengineering/r.pkg.template/assets/12943682/9dc2aa08-5e69-4f70-919e-7e457be7a4df)

Take a look at the timings for `valgrind`. It took 4 times longer than the median timing of all the platforms. If the minutes consumption are in concern - maybe we can remove it? WDYT?

Reply:
AFAIK, GitHub minutes consumption is not a problem for public repositories (they are free) but we can remove any irrelevant checks, especially if they take a long time.